packages/aws-cdk-lib/custom-resources/lib/provider-framework/waiter-state-machine.ts

```diff
@@ -75,23 +123,61 @@ export class WaiterStateMachine extends Construct {
       },
     });
 
-    const resource = new CfnResource(this, 'Resource', {
-      type: 'AWS::StepFunctions::StateMachine',
-      properties: {
-        DefinitionString: definition,
-        RoleArn: role.roleArn,
-      },
+    const resource = new CfnStateMachine(this, 'Resource', {
+      definitionString: definition,
+      roleArn: role.roleArn,
+      loggingConfiguration: this.renderLoggingConfiguration(role, props.logOptions, props.disableLogging),
     });
     resource.node.addDependency(role);
 
     this.stateMachineArn = resource.ref;
   }
 
+  /**
+   * Grant the given identity permissions on StartExecution of the state machine.
+   */
   public grantStartExecution(identity: IGrantable) {
     return Grant.addToPrincipal({
       grantee: identity,
       actions: ['states:StartExecution'],
       resourceArns: [this.stateMachineArn],
     });
   }
+
+  private renderLoggingConfiguration(
+    role: Role,
+    logOptions?: LogOptions,
+    disableLogging?: boolean,
+  ): CfnStateMachine.LoggingConfigurationProperty | undefined {
+    if (disableLogging) return undefined;
+
+    // https://docs.aws.amazon.com/step-functions/latest/dg/cw-logs.html#cloudwatch-iam-policy
```

Comment:
Thanks for this link. Let's also copy in the important piece of information from this link: 

```ts
// You need to specify `*` in the Resource field because CloudWatch API actions, such as CreateLogDelivery and DescribeLogGroups, don't support Resource types defined by Amazon CloudWatch Logs.
```